src/Microsoft.TemplateEngine.Abstractions/Installer/InstallResult.cs

```diff
@@ -10,32 +12,41 @@ namespace Microsoft.TemplateEngine.Abstractions.Installer
     /// </summary>
     public sealed class InstallResult : InstallerOperationResult
     {
-        private InstallResult(InstallRequest request, IManagedTemplatePackage templatePackage)
+        private InstallResult(InstallRequest request, IManagedTemplatePackage templatePackage, Dictionary<Uri, int>? vulnerabilities = null)
             : base(templatePackage)
         {
             InstallRequest = request;
+            Vulnerabilities = vulnerabilities;
         }
 
-        private InstallResult(InstallRequest request, InstallerErrorCode error, string errorMessage)
+        private InstallResult(InstallRequest request, InstallerErrorCode error, string errorMessage, Dictionary<Uri, int>? vulnerabilities)
              : base(error, errorMessage)
         {
             InstallRequest = request;
+            Vulnerabilities = vulnerabilities;
         }
 
         /// <summary>
         /// <see cref="InstallRequest"/> processed by <see cref="IInstaller.InstallAsync"/> operation.
         /// </summary>
         public InstallRequest InstallRequest { get; private set; }
 
+        /// <summary>
+        /// Vulnerabilities from installed package.
+        /// </summary>
+        /// It is a dictionary, as we don't want to add the NuGet Api package to this project.
+        public Dictionary<Uri, int>? Vulnerabilities { get; private set; }
```

Comment:
```suggestion
        public IReadOnlyDictionary<Uri, int>? Vulnerabilities { get; private set; }
```

use read-only interfaces when change to collection is not intended